http_parser.h

```diff
@@ -304,6 +304,7 @@ struct http_parser {
   unsigned int index : 5;        /* index into current matcher */
   unsigned int extra_flags : 2;
   unsigned int lenient_http_headers : 1;
+  unsigned int allow_cl_te: 1;
```

Comment:
All the other fields above add up to 32 bits, this one additional bit pushes into the next 32 bits, and causes the size and offsets of `struct http_parser` to change. That's not good.

You may be able to take one bit from `extra_flags`, it looks like only one of those two bits is actually used, for `F_TRANSFER_ENCODING`. It's awkward because it was intended to support more "overflow" `flags`, but with this change would only support the one. It's also awkward because this is in the "PRIVATE" section of the struct, and library users are intended to set it directly. It looks like there aren't any great options here.

Reply:
>It's also awkward because this is in the "PRIVATE" section of the struct, and library users are intended to set it directly. It looks like there aren't any great options here.
Yes, it is. Was trying to make it similar to `lenient_http_headers` in initial implementation.
What do you think if I extend `settings` structure to allow library users to configure behavior there?

Reply:
@ploxiln updated PR, could you take a look again, please?

Reply:
Can you move the definition of `allow_cl_te` before `lenient_http_headers`? The position of the latter shouldn't change.

It also needs a comment explaining what it does and `allow_cl_te` feels a bit too cryptic, a more self-descriptive name would be better. Minor style issue: space before `:`

It's okay to reduce `extra_flags` to a 1-bit field but can you rename it to e.g. `uses_transfer_encoding` and update the code accordingly?

The `F_TRANSFER_ENCODING` flag will be unused at that point and it should be safe to remove. It's technically an API break but it's not like you can use it meaningfully now anyway.

Reply:
updated the PR